Gemfile

```diff
@@ -147,6 +147,7 @@ def override_gem(name, *args)
     dependencies.delete(dependency)
 
     gem name, *args
+    Bundler::UI::Shell.new.warn "Caution, using override_gem for #{name} with args: #{args.inspect}, review Gemfile.dev.rb if this is not desired!"
```

Comment:
* I can do a `.tap` on the result of the `gem` method to print this which would solve the problems of printing it on failure and not losing the result of `gem`, etc.
* It's easy enough to do a `unless ENV['RAILS_ENV'] == "production"`
* I don't think it's intrusive to see this message (or the simpler one others are suggesting) if you're using the `override_gem` normally...therefore I'm not sure why you'd want to opt-out